src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -1078,9 +1085,9 @@ internal bool HasIsByRefLikeAttribute(EntityHandle token)
             }
 
             info = FindTargetAttribute(token, AttributeDescription.ObsoleteAttribute);
-            if (info.HasValue)
+            if (info.HasValue && decoder.GetCustomAttribute(info.Handle, out TypedConstant[] positionalArgs, out KeyValuePair<string, TypedConstant>[] namedArgs, out bool[] namedArgIsProperty))
```

Comment:
>decoder.GetCustomAttribute [](start = 33, length = 26)

It looks like this isn't quite what I was suggesting: "take an instance of MetadataDecoder and leverage helpers it has to crack individual named args" I.e. we should try not using the decoder to crack constructor arguments and build any arrays. Crack arguments as we do today, then crack named arguments one by one by using decoder.

Reply:
Thanks for the clarification, will fix.

Reply:
Done